From 5338f69a0a3dcc8527d81ca5f936b0e066a3d7f7 Mon Sep 17 00:00:00 2001
From: Karl Schultz <karl@lunarg.com>
Date: Thu, 3 Nov 2016 12:31:41 -0600
Subject: [PATCH] layers: GH1115 Add missing VK_LAYER_EXPORT decorations

Functions in the vk_layer_extension_utils file were
not getting exported correctly.

Change-Id: If1bf6456cb8e79a1b6292ea76bd72e3e89a2b430
---
 layers/vk_layer_extension_utils.cpp | 8 ++++----
 layers/vk_layer_extension_utils.h   | 8 ++++----
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/layers/vk_layer_extension_utils.cpp b/layers/vk_layer_extension_utils.cpp
index bd778e9..4c1e396 100644
--- a/layers/vk_layer_extension_utils.cpp
+++ b/layers/vk_layer_extension_utils.cpp
@@ -27,8 +27,8 @@
  * This file contains utility functions for layers
  */
 
-VkResult util_GetExtensionProperties(const uint32_t count, const VkExtensionProperties *layer_extensions, uint32_t *pCount,
-                                     VkExtensionProperties *pProperties) {
+VK_LAYER_EXPORT VkResult util_GetExtensionProperties(const uint32_t count, const VkExtensionProperties *layer_extensions,
+                                                     uint32_t *pCount, VkExtensionProperties *pProperties) {
     uint32_t copy_size;
 
     if (pProperties == NULL || layer_extensions == NULL) {
@@ -46,8 +46,8 @@ VkResult util_GetExtensionProperties(const uint32_t count, const VkExtensionProp
     return VK_SUCCESS;
 }
 
-VkResult util_GetLayerProperties(const uint32_t count, const VkLayerProperties *layer_properties, uint32_t *pCount,
-                                 VkLayerProperties *pProperties) {
+VK_LAYER_EXPORT VkResult util_GetLayerProperties(const uint32_t count, const VkLayerProperties *layer_properties, uint32_t *pCount,
+                                                 VkLayerProperties *pProperties) {
     uint32_t copy_size;
 
     if (pProperties == NULL || layer_properties == NULL) {
diff --git a/layers/vk_layer_extension_utils.h b/layers/vk_layer_extension_utils.h
index 0f249b7..b593768 100644
--- a/layers/vk_layer_extension_utils.h
+++ b/layers/vk_layer_extension_utils.h
@@ -30,11 +30,11 @@
  */
 extern "C" {
 
-VkResult util_GetExtensionProperties(const uint32_t count, const VkExtensionProperties *layer_extensions, uint32_t *pCount,
-                                     VkExtensionProperties *pProperties);
+VK_LAYER_EXPORT VkResult util_GetExtensionProperties(const uint32_t count, const VkExtensionProperties *layer_extensions,
+                                                     uint32_t *pCount, VkExtensionProperties *pProperties);
 
-VkResult util_GetLayerProperties(const uint32_t count, const VkLayerProperties *layer_properties, uint32_t *pCount,
-                                 VkLayerProperties *pProperties);
+VK_LAYER_EXPORT VkResult util_GetLayerProperties(const uint32_t count, const VkLayerProperties *layer_properties, uint32_t *pCount,
+                                                 VkLayerProperties *pProperties);
 
 } // extern "C"
 #endif // LAYER_EXTENSION_UTILS_H
